screens/Menu/CourseSchedule.js

```diff
@@ -23,53 +24,127 @@ import PropTypes from "prop-types";
  * @param  {} navigation props.navigation is the name of the object from Navigator library
  */
 function CourseSchedule(props) {
+    const [accessToken, setAccessToken] = React.useState(null);
+    const [selectedCalendarId, setSelectedCalendarId] = React.useState(null);
+    const [calendarEventsList, setCalendarEventsList] = React.useState(null);
+    const [switchVal, setSwitchVal] = React.useState("false");
+
+    //TODO Show only events after current date
+    let currentDate = new Date();
+
+    const getCalendarId = async () => {
+        let calendarId = await AsyncStorage.getItem("calendarId");
+        setSelectedCalendarId(calendarId);
+    };
+
+    const getAccessToken = async () => {
+        let accessToken = await AsyncStorage.getItem("accessToken");
+        setAccessToken(accessToken);
+    };
 
-     /**
+    const getSwitchValue = async () => {
+        let switchValue = await AsyncStorage.getItem('switchVal');
+        setSwitchVal(switchValue);
+    };
+
+    const getCalendarEvents = async () => {
+        let calendarEvents = await fetch('https://www.googleapis.com/calendar/v3/calendars/' + selectedCalendarId + '/events', {
+            headers: { Authorization: `Bearer ${accessToken}` }, singleEvents: true, orderBy: 'startTime'
+        });
+        let resp = await calendarEvents.json();
+        setCalendarEventsList(getFilteredGoogleCalendarEvents(resp));
+    }
+
+    const getFilteredGoogleCalendarEvents = (resp) => {
+        var filteredList = resp.items.map(element => {
+            return { id: element.id, summary: element.summary, description: element.description, location: element.location };
+        });
+        return filteredList;
+    };
+
+    const CourseScheduleScreen = true;
+
+    /**
      * The method will slide the side menu from the right side of the screen
      * @param  {} =>{props.navigation.openDrawer(
-     */
+    */
     const goToMenu = () => {
         props.navigation.openDrawer();
     };
 
-    
-    /**
-     * The method will let us navigate to the CourseSchedule screen
-     * @param  {} =>{props.navigation.navigate("CourseScheduleDetails"
-     */
-    const goToCourseSchedule = () => {
-        props.navigation.navigate("CourseScheduleDetails");
-    };
+    const goToDoubleSearch = (item) => {
+        props.navigation.navigate("DoubleSearch", { CourseScheduleScreen: CourseScheduleScreen, CourseScheduleLocation: item.location });
+    }
 
-    // TODO: Static data for now just to layout the SectionList 
-    const data = [{
-        title: "Wed, Feb 5",
-        data: ["SOEN 357 Lecture", "SOEN 357 Tutorial", "SOEN 345 Lecture"
-        ]
-    }];
+    useEffect(() => {
+        const intervalId = setInterval(() => {
```

Comment:
instead of refreshing every 1ms, we should add a scroll up to refresh in order to avoid intervals

Reply:
I approve